### PR TITLE
Search_Algorithm

### LINKED ISSUE #36: Searching Algorithms
https://www.studytonight.com/data-structures/introduction-to-data-structures

Read this full section:
image

Then after that, in Ruby: 
- [x] Linear Search
- [x] Binary Search

### PR DESCRIPTION
**Pull request by Anishluke92**

### CODE REVIEW

File: linear_search_algorithm.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+def linearSearch(array,target)
+
+    i = 0 
+    while i < array.length 
+        if array[i] == target
```
Comment (danielpaul) [inline linear_search_algorithm.rb:5]: you can also do `array.each_with_index do |value, index|`

Comment (danielpaul) [reply linear_search_algorithm.rb:5]: so linear search is basically just looping through and finding the number if it is there

File: Binary_search.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+def binary_search(array, target)
```
Comment (danielpaul) [inline Binary_search.rb:1]: if they ask you to do this in a whiteboard, can you do this without looking at anything?

**Review by danielpaul — CHANGES_REQUESTED** [review]

File: linear_search_algorithm.rb (hunk 1 above)
Comment (Anishluke92) [reply linear_search_algorithm.rb:5]: yes of course we can use it, but I have been using .each for most of my problem solving. thought i  could use  while for this. Yes linear search is just an simple iteration search algorithm were we iterate through every element on the  list until we find the match to the target.

File: Binary_search.rb (hunk 2 above)
Comment (Anishluke92) [reply Binary_search.rb:1]: Before i started coding this, i solved this using paper and pen for better understanding. Honestly reading the algorithm or coding it didn't given me much understanding of how it works. so i had to use paper and pen and  solved which actually helped me understand how this algorithm actually works. I'm using this way of understanding on sorting algorithms also

**Review by danielpaul — APPROVED** [review]

**Pull request merged**